app/src/main/java/com/example/chaika/ui/viewModels/TripViewModel.kt

```
@@ -0,0 +1,157 @@
+
+    fun loadHistory() {
+        loadHistoryData()
+        getHistory()
+    }
+
+    private fun loadHistoryData() {
+        viewModelScope.launch {
+            _isLoading.value = true
+            try {
+                fetchAndSaveHistoryUseCase()
+            } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)